src/gui/render/graphics_shaders/individual_int_renderer.hpp

```diff
@@ -0,0 +1,406 @@
+// -*- lsst-c++ -*-
+/*
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ */
+
+/**
+ * @file individual_int_renderer.hpp
+ *
+ * @brief Defines MainRender class.
+ *
+ * @ingroup GUI  MODELS
+ *
+ */
+#pragma once
+
+#include "../../../util/files.hpp"
+#include "../../handler.hpp"
+#include "../../scene/controls.hpp"
+#include "../../shader.hpp"
+#include "../graphics_data/individual_int.hpp"
+#include "gui_render_types.hpp"
+
+#include <GL/glew.h>
+#include <GLFW/glfw3.h>
+#include <glm/glm.hpp>
+#include <glm/gtc/matrix_transform.hpp>
+
+#include <memory>
+#include <vector>
+
+namespace gui {
+
+namespace models {
+
+/**
+ * @brief Renders the meshes to the screen
+ *
+ * @details IndividualIntRenderer renders the meshes given to the screen.
+ * This class handles IndividualInt data structures and classes that derive from
+ * IndividualInt.
+ */
+template <class T>
+class IndividualIntRenderer :
+    public render_to::frame_buffer_multisample,
+    public render_to::frame_buffer,
+    public render_to::shadow_map {
+ protected:
+    GLuint program_id_render_; // ID of render program
+    GLuint program_id_shadow_; // ID of shadow program
+
+    GLuint matrix_id_render_;      // uniform ID of transform matrix
+    GLuint view_matrix_id_render_; // ID of view projection matrix for indexed meshes
+    GLuint depth_bias_id_render_;  // ID of depth projection matrix for indexed meshes
+    GLuint shadow_map_id_render_;  // ID of the shadow map for indexed meshes
+    GLuint color_map_id_render_;   // ID of the color map for indexed meshes
+    GLuint light_direction_id_render_; // ID of the light direction uniform for indexed
+
+    GLuint depth_bias_id_shadow_; // ID of depth projection matrix for indexed meshes
+    // ------ the below are added to the class ------
+    GLuint depth_texture_;              // ID of the shadow depth texture
+    glm::vec3 light_direction_;         // direction of sun light
+    glm::mat4 depth_projection_matrix_; // projection matrix of the light source
+    glm::mat4 depth_view_matrix_; // convert a point in world space to depth in light
+    //  light direction
+    std::vector<std::shared_ptr<T>> meshes_;
+
+ public:
+    /**
+     * @brief Construct a new Main Renderer object
+     *
+     * @param ShaderHandler
+     */
+    IndividualIntRenderer(ShaderHandler shader_handler = ShaderHandler());
```

Comment:
```suggestion
    explicit IndividualIntRenderer(ShaderHandler shader_handler = ShaderHandler());
```

Reply:
what? why nothing can be converted to a shader_handler?